configure.ac

```diff
@@ -187,7 +187,20 @@ if test a"$clamav" = "ayes"; then
     AC_DEFINE(HAVE_LIBCLAMAV_095,1,[Define HAVE_LIBCLAMAV_095 if have clamav 0.95.x or newer])
     AC_MSG_RESULT(yes),
     )
+
+    #
+    # clamav dropped CL_SCAN_HEURISTIC_ENCRYPTED in 0.101 replacing it with
+    # CL_SCAN_HEURISTIC_ENCRYPTED_ARCHIVE and CL_SCAN_HEURISTIC_ENCRYPTED_DOC
     # restore flags  / clamav tests
+    AC_MSG_CHECKING([for clamav 0.101 or newer])
+    AC_TRY_COMPILE(
+    [#include <clamav.h>],
+    [int test1=CL_SCAN_HEURISTIC_ENCRYPTED_ARCHIVE;
+     int test2=CL_SCAN_HEURISTIC_ENCRYPTED_DOC;],
+    AC_DEFINE(HAVE_LIBCLAMAV_0101,1,[Define HAVE_LIBCLAMAV_0101 if have clamav 0.101.x or newer])
+    AC_MSG_RESULT(yes),
+    AC_MSG_RESULT(no),
+    )
     CFLAGS=$OLD_CFLAGS
```

Comment:
It is better to not detect and define the HAVE_LIBCLAMAV_0101 but instead detect if the "struct cl_scan_options" is defined and define a HAVE_CL_SCAN_OPTIONS. Then inside  clamav_post_init method use:
`#ifdef  HAVE_CL_SCAN_OPTIONS`
`     memset(&CLAMSCAN_OPTIONS, 1, sizeof(CLAMSCAN_OPTIONS));`
`     CLAMSCAN_OPTIONS.parse = ~0;`
`...`
`#else `
`#if defined(CL_SCAN_BLOCKENCRYPTED)`
`     if (CLAMAV_BLOCKENCRYPTED)`
`         CLAMSCAN_OPTIONS |= CL_SCAN_BLOCKENCRYPTED;`
`#endif`
`...`
`#endif`

Reply:
Which options are you referring to? Are you wanting to separate encrypted archive and encrypted doc? Are you talking about the parse options? 

    CL_SCAN_PARSE_ARCHIVE This flag enables transparent scanning of various archive formats.
    CL_SCAN_PARSE_ELF Enable support for ELF files.
    CL_SCAN_PARSE_PDF Enables scanning within PDF files.
    CL_SCAN_PARSE_SWF Enables scanning within SWF files, notably compressed SWF.
    CL_SCAN_PARSE_HWP Enables scanning of Hangul Word Processor (HWP) files.
    CL_SCAN_PARSE_XMLDOCS Enables scanning of XML-formatted documents (e.g. Word, Excel, Powerpoint, HWP).
    CL_SCAN_PARSE_MAIL Enable support for mail files.
    CL_SCAN_PARSE_OLE2 Enables support for OLE2 containers (used by MS Office and .msi files).
    CL_SCAN_PARSE_HTML This flag enables HTML normalisation (including ScrEnc decryption).
    CL_SCAN_PARSE_PE This flag enables deep scanning of Portable Executable files and allows libclamav to unpack executables compressed with run-time unpackers.

Right now, the parse options are just all set, which was the simple recommendation in the documentation at https://github.com/Cisco-Talos/clamav-devel/blob/dev/0.101/docs/UserManual/libclamav.md#data-scan-functions

I am testing a new patch that makes sure that if virusonfailure is set, that the appropriate option is also set.

Reply:
The file types it is better to be all enabled. 
Then the virus_scan.ScanFileTypes can be used to control the filetypes someone want to scan for viruses.
So there is not any need to set/unset them.